### PR TITLE
Bug 1814280: UPSTREAM 278: Include a nil check for the snapshotContent

### PR DESCRIPTION
**Pull request by huffmanca**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This is a backport of https://github.com/kubernetes-csi/external-snapshotter/pull/278 . It addresses an issue where the snapshotController can crash when checking snapshot Finalizers if the VolumeSnapshotContent is nil. 

**Which issue(s) this PR fixes**:
Fixes BZ1814280

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/common-controller/snapshot_controller.go [hunk 1]
```diff
@@ -21,9 +21,9 @@ import (
 	"strings"
 	"time"
 
-	crdv1 "github.com/kubernetes-csi/external-snapshotter/pkg/apis/volumesnapshot/v1beta1"
-	"github.com/kubernetes-csi/external-snapshotter/pkg/utils"
-	"k8s.io/api/core/v1"
+	crdv1 "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	"github.com/kubernetes-csi/external-snapshotter/v2/pkg/utils"
```
Comment (bertinatto) [inline pkg/common-controller/snapshot_controller.go]: These two lines above change the package version used in this file. This change isn't part of the upstream commit, I don't think we want those, do we?

Comment (jsafrane) [reply pkg/common-controller/snapshot_controller.go]: No, we don't want this.

Comment (huffmanca) [reply pkg/common-controller/snapshot_controller.go]: I removed this change, and the tests are passing as expected now. This one should be good for review and cherry-picking once merged.

**Comment by huffmanca** [comment]
Closing this, as the PR was reverted.

**Comment by huffmanca** [comment]
Reopening, as Xing didn't revert the PR. The original branch has been updated to include the second commit which resolves this issue.

**Pull request merged**